networking/networking.go

```diff
@@ -183,6 +243,43 @@ func (n *Networking) GetDefaultHostIP() (net.IP, error) {
 	return n.nets[len(n.nets)-1].runtime.HostIP, nil
 }
 
+// GetIfacesByIP searches for and returns the interfaces with the given IP
+// Disregards the subnet mask since not every net.IP object contains
+// On success it will return the list of found interfaces
+func (n *Networking) GetIfacesByIP(ifaceIP net.IP) ([]net.Interface, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+
+	searchAddr := strings.Split(ifaceIP.String(), "/")[0]
+	resultInterfaces := make([]net.Interface, 0)
+
+	for _, iface := range ifaces {
+		if iface.Flags&net.FlagLoopback != 0 {
+			continue
+		}
+
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return nil, errwrap.Wrap(fmt.Errorf("cannot get addresses for interface %v", iface.Name), err)
+		}
+
+		for _, addr := range addrs {
+			currentAddr := strings.Split(addr.String(), "/")[0]
+			if searchAddr == currentAddr {
+				resultInterfaces = append(resultInterfaces, iface)
```

Comment:
It could be more than one interface for the way [ptp is currently implemented](https://github.com/appc/cni/blob/137dc8a673e6ed38c49e3b8d1cf03858271db9a8/plugins/main/ptp/ptp.go#L48).